### PR TITLE
Fix incorrect examples tests

### PR DESCRIPTION
**Pull request by mhorod**

### CODE REVIEW

File: src/main/kotlin/cacophony/semantic/names/NameResolver.kt [hunk 1]
```diff
@@ -157,6 +157,16 @@ fun resolveNames(root: AST, diagnostics: Diagnostics): NameResolutionResult {
...
+                    .filter { it.value.size > 1 }
+                    .values
+                    .flatten()
+                    .forEach { argNode ->
+                        diagnostics.report(NRDiagnostics.DuplicatedFunctionArgument(argNode.identifier), argNode.range)
+                    }
+
                 // Open new block to make arguments visible in function body, but not after
                 // the whole function declaration.
                 symbolsTable.open()
```
Comment (mhorod) [inline src/main/kotlin/cacophony/semantic/names/NameResolver.kt:172]: I'm not sure if we should continue analyzing function when there's an error with the arguments 🤔

File: src/test/kotlin/cacophony/examples/ExamplesTest.kt [hunk 2]
```diff
@@ -77,7 +76,6 @@ class ExamplesTest {
     // TODO: Fix duplicated function params
     @ParameterizedTest
     @MethodSource("incorrectExamples")
-    @Disabled
```
Comment (jjaworska) [inline src/test/kotlin/cacophony/examples/ExamplesTest.kt:80]: :confetti_ball:

File: src/test/kotlin/cacophony/semantic/names/NameResolverTest.kt [hunk 3]
```diff
@@ -1214,5 +1214,41 @@ class NameResolverTest {
...
+        @Test
+        fun `error when finds duplicated argument names`() {
+            // let f = [x: Int, x: Bool] -> Int => 0
+
+            // given
+            val firstArgumentRange = Location(1) to Location(1)
+            val secondArgumentRange = Location(2) to Location(2)
+            val ast =
+                block(
+                    functionDeclaration(
```
Comment (jjaworska) [inline src/test/kotlin/cacophony/semantic/names/NameResolverTest.kt:1227]: nit, but if body is `lit(0)`, maybe let's make this a intFunctionDeclaration

**Review by jjaworska — APPROVED** [review]
Looks great! Just one question, a couple of small remarks about comments and one thing up to discussion

File: src/test/kotlin/cacophony/semantic/names/NameResolverTest.kt (hunk 3 above)
Comment (mhorod) [reply src/test/kotlin/cacophony/semantic/names/NameResolverTest.kt]: good catch

File: src/test/kotlin/cacophony/TestAstUtils.kt [hunk 4]
```diff
@@ -19,10 +26,51 @@ fun functionDeclaration(identifier: String, body: Expression) =
...
+        returnType,
+        body,
+    )
+
+/**
+ * Declares a function of type [] -> Int
+ */
+fun intFunctionDeclaration(identifier: String, body: Expression) = functionDeclaration(identifier, body, intType())
+
+fun intFunctionDeclaration(identifier: String, args: List<Definition.FunctionArgument>, body: Expression) =
```
[the remark this answers is not in the document]
Comment (mhorod) [reply src/test/kotlin/cacophony/TestAstUtils.kt:36]: fixed, also I changed `functionDeclaration` to `unitFunctionDeclaration` for consistency

File: src/main/kotlin/cacophony/pipeline/CacophonyPipeline.kt [hunk 5]
```diff
@@ -162,6 +163,7 @@ class CacophonyPipeline(
 
     private fun analyzeFunctions(ast: AST): FunctionAnalysisResult {
         val resolvedFunctions = resolveOverloads(ast)
+        checkTypes(ast, resolvedFunctions)
```
[the remark this answers is not in the document]
Comment (mhorod) [reply src/main/kotlin/cacophony/pipeline/CacophonyPipeline.kt:166]: This one is currently used in `pipeline.process` which is used in `ExampleTests`.  Removing this line actually causes incorrect tests to fail due to typechecker not being run. 

We'll have to clean up the pipeline eventually 😄

File: src/test/kotlin/cacophony/controlflow/generation/CallTest.kt [hunk 6]
```diff
@@ -2,23 +2,22 @@ package cacophony.controlflow.generation
...
 class CallTest {
     @Test
     fun `call sequence for a function without parameters is correctly generated when the call is used as a value`() {
         // given
-        val calleeDef = functionDeclaration("callee", lit(1))
-        val callerDef = functionDeclaration("caller", call("callee"))
+        val calleeDef = intFunctionDeclaration("callee", lit(1))
+        val callerDef = intFunctionDeclaration("caller", call("callee"))
         /*
          * let callee = [] -> Int => 1;
```
[the remark this answers is not in the document]
Comment (jjaworska) [reply src/test/kotlin/cacophony/controlflow/generation/CallTest.kt:14]: Wooow, sorry, mb :laughing:

File: src/main/kotlin/cacophony/semantic/names/NameResolver.kt (hunk 1 above)
Comment (mhorod) [reply src/main/kotlin/cacophony/semantic/names/NameResolver.kt]: Changed it to be fatal, we can refactor this after implementing structs and seeing how we handle them

**Pull request merged**